src/renovate.ts

```diff
@@ -0,0 +1,53 @@
+import Docker from './docker';
+import child from 'child_process';
+import fs from 'fs';
+import path from 'path';
+
+class Renovate {
+  private configFileEnv = 'RENOVATE_CONFIG_FILE';
+  private tokenEnv = 'RENOVATE_TOKEN';
+  private configFileMountDir = '/github-action';
+
+  private configFile: string;
+  private docker: Docker;
+
+  constructor(configFile: string, private token: string) {
+    this.configFile = path.resolve(configFile);
+
+    this.validateArguments();
+
+    this.docker = new Docker();
+  }
+
+  runDockerContainer(): void {
+    const commandArguments = [
+      '--rm',
+      `--env ${this.configFileEnv}='${this.configFileMountPath()}'`,
+      `--env ${this.tokenEnv}='${this.token}'`,
+      `--volume ${this.configFile}:${this.configFileMountPath()}`,
+      this.docker.image(),
+    ];
+    const command = `docker run ${commandArguments.join(' ')}`;
+    // console.log(commandArguments, command);
+
+    child.execSync(command, { stdio: 'inherit' });
```

Comment:
I would prefer to use `@actions/exec`

https://github.com/renovatebot/internal-tools/blob/5c8f6cdc0e60da73b5d12c5d5e1b54cc071265d7/src/util.ts#L20-L51